api/client/gro_client.py

```diff
@@ -53,12 +53,15 @@ def __init__(self, api_host, access_token):
     ###
     def get_df(self):
         """Get the content of all data series in a Pandas frame."""
-        frames = [self._data_frame]
         while self._data_series_queue:
-            frames.append(pandas.DataFrame(data=self.get_data_points(
-                **self._data_series_queue.pop())))
-        if len(frames) > 1:
-            self._data_frame = pandas.concat(frames)
+            tmp = pandas.DataFrame(
+                data=self.get_data_points(**self._data_series_queue.pop()))
+            tmp.end_date = pandas.to_datetime(tmp.end_date)
```

Comment:
end_date and start_date members won't be available here if no data points are returned.

Reply:
Made it conditional.

Reply:
lgtm